editor/editor_properties_array_dict.cpp

```diff
@@ -504,6 +505,56 @@ bool EditorPropertyArray::_is_drop_valid(const Dictionary &p_drag_data) const {
 		return true;
 	}
 
+	if (drop_type == "nodes") {
+		Array node_paths = drag_data["nodes"];
+
+		PackedStringArray allowed_subtype_array;
+		if (allowed_type == "NodePath") {
+			if (subtype_hint_string == "NodePath") {
+				return true;
+			} else {
+				for (int j = 0; j < subtype_hint_string.get_slice_count(","); j++) {
+					String ast = subtype_hint_string.get_slice(",", j).strip_edges();
+					allowed_subtype_array.append(ast);
+				}
+			}
+		}
+
+		TypedArray<bool> is_drop_allowed;
```

Comment:
That's some overkill approach...
If you need a single node to fail, you could just break when `is_parent_class()` returns false. There is no reason to continue the loop.

Reply:
Let's say I have `@export_node_path("Node2D", "Node3D")` as allowed subtypes and then I drop a CharacterBody3D. `is_parent_class()` will return false the first time, so if I break there it will not allow to drop it even tho it should be possible since the second allowed type is Node3D and CharacterBody3D inherits it.
I need to be able to overwrite it because I can't know in advance the order of things, or I would have the issue mentioned before. It's already a bit too convoluted for my brain lol; I don't see how I could make it simpler honestly. I tried and there was always an issue one way or another, that's why I ended up with this solution which works. But it's highly possible that there's a better approach, just that my brain is unable to handle it ... Maybe tomorrow it will be clearer.